tests/run/test_named_expressions.py

```diff
@@ -0,0 +1,485 @@
+# copied from cpython with minimal modifications (mainly exec->cython.inline, and a exception strings)
+
+import os
+import unittest
+import cython
+from Cython.Build.Inline import cython_inline
+from Cython.Compiler.Main import CompileError
+
+if cython.compiled:
+    def exec_(code, locals_, globals_):
```

Comment:
In other copied Python tests, I just overwrite `eval()`, `exec()` and/or `compile()`. That reduces the amount of modifications in the test functions below. There _will_ be an update to this test file at some point in the future, and then it's often easier to copy the new file over and undo all local changes, than to copy over the CPython changes one by one.